### PR TITLE
api: lineage

### PR DESCRIPTION
**Pull request by jirikuncar**
Local development:

0. edit `config.py` so it doesn't download the OIDC public key
1. Run Apache Jena with your data: `docker run -e ADMIN_PASSWORD=admin 3030:3030 -v $(pwd)/fuseki:/fuseki -v $(pwd)/staging:/staging stain/jena-fuseki`
2. Launch the gateway `GITLAB_URL=https://renkulab.io/gitlab GITLAB_CLIENT_SECRET=aaaa JUPYTERHUB_CLIENT_SECRET=aaaa OIDC_CLIENT_SECRET=aaa GATEWAY_SECRET_KEY=aaaa QUART_DEBUG=1 QUART_APP=app:app quart run`

--
cc @jachro @ciyer 

--- 
addresses #71

### CODE REVIEW

**Comment by rokroskar** [comment]
I tried to run the image and I get:

```
Running on http://0.0.0.0:5000 (CTRL + C to quit)
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/hypercorn/utils.py", line 55, in load_application
    module = import_module(import_name)
  File "/usr/local/lib/python3.7/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 965, in _find_and_load_unlocked
ModuleNotFoundError: No module named 'run'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/bin/hypercorn", line 11, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.7/site-packages/hypercorn/__main__.py", line 165, in main
    run(config)
  File "/usr/local/lib/python3.7/site-packages/hypercorn/run.py", line 34, in run
    worker_func(config)
  File "/usr/local/lib/python3.7/site-packages/hypercorn/asyncio/run.py", line 192, in asyncio_worker
    app = load_application(config.application_path)
  File "/usr/local/lib/python3.7/site-packages/hypercorn/utils.py", line 58, in load_application
    raise NoAppException()
hypercorn.utils.NoAppException
```

File: app/blueprints/graph.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2018 - Swiss Data Science Center (SDSC)
```
Comment (rokroskar) [inline app/blueprints/graph.py]:
```suggestion
# Copyright 2019 - Swiss Data Science Center (SDSC)
```

**Comment by jirikuncar** [comment]
> I tried to run the image and I get:
> 
> ```
> Running on http://0.0.0.0:5000 (CTRL + C to quit)
> Traceback (most recent call last):
>   File "/usr/local/lib/python3.7/site-packages/hypercorn/utils.py", line 55, in load_application
>     module = import_module(import_name)
>   File "/usr/local/lib/python3.7/importlib/__init__.py", line 127, in import_module
>     return _bootstrap._gcd_import(name[level:], package, level)
>   File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
>   File "<frozen importlib._bootstrap>", line 983, in _find_and_load
>   File "<frozen importlib._bootstrap>", line 965, in _find_and_load_unlocked
> ModuleNotFoundError: No module named 'run'
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last):
>   File "/usr/local/bin/hypercorn", line 11, in <module>
>     sys.exit(main())
>   File "/usr/local/lib/python3.7/site-packages/hypercorn/__main__.py", line 165, in main
>     run(config)
>   File "/usr/local/lib/python3.7/site-packages/hypercorn/run.py", line 34, in run
>     worker_func(config)
>   File "/usr/local/lib/python3.7/site-packages/hypercorn/asyncio/run.py", line 192, in asyncio_worker
>     app = load_application(config.application_path)
>   File "/usr/local/lib/python3.7/site-packages/hypercorn/utils.py", line 58, in load_application
>     raise NoAppException()
> hypercorn.utils.NoAppException
> ```

@rokroskar Have you run it inside `pipenv shell`?

**Comment by jirikuncar** [comment]
@rokroskar I have fixed the Dockerfile.

**Comment by rokroskar** [comment]
The image runs now, but it never looks healthy to k8s. Has the health endpoint changed?

File: app/__init__.py [hunk 2]
```diff
@@ -17,39 +17,69 @@
...
+blueprints = (
+    graph.blueprint,
+    gitlab_auth.blueprint,
+    jupyterhub_auth.blueprint,
+    proxy.blueprint,
+    web.blueprint,
+)
+
+
+@app.route('/health', methods=['GET'])
```
Comment (jirikuncar) [inline app/__init__.py]: @rokroskar the `/health` endpoint is in the root now.

**Comment by rokroskar** [comment]
@jirikuncar looks like 6c85fe9 broke the tests...

File: helm-chart/renku-gateway/templates/deployment.yaml [hunk 3]
```diff
@@ -78,6 +78,12 @@ spec:
                 secretKeyRef:
                   name: {{ template "gateway.fullname" . }}
                   key: oidcClientSecret
+            - name: SPARQL_ENDPOINT
+              value: {{ .Values.graph.sparql.endpoint | default (printf "%s://jena.%s/renku/sparql" (include "gateway.protocol" .) .Values.global.renku.domain ) | quote }}
+            - name: SPARQL_USERNAME
+              value: {{ required "Please specify a username for the SPARQL endpoint." .Values.graph.sparql.username | quote }}
+            - name: SPARQL_PASSWORD
+              value: {{ required "Please specify a password for the SPARQL endpoint user." .Values.graph.sparql.password | quote }}
```
Comment (jirikuncar) [inline helm-chart/renku-gateway/templates/deployment.yaml]: 👍

**Pull request merged**